pkg/foundation/cerrors/fatal_test.go

```diff
@@ -0,0 +1,78 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cerrors_test
+
+import (
+	"testing"
+
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+)
+
+func TestNewFatalError(t *testing.T) {
+	err := cerrors.New("test error")
+	fatalErr := cerrors.NewFatalError(err)
+
+	if fatalErr.Error() != err.Error() {
+		t.Errorf("expected error message to be %s, got %s", err.Error(), fatalErr.Error())
+	}
+}
+
+func TestIsFatalError(t *testing.T) {
+	err := cerrors.New("test error")
+
+	testCases := []struct {
+		name string
+		err  error
+		want bool
+	}{
+		{
+			name: "FatalError",
+			err:  cerrors.NewFatalError(err),
+			want: true,
+		},
+		{
+			name: "No Fatal Error",
+			err:  cerrors.New("test error"),
+			want: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			got := cerrors.IsFatalError(tc.err)
+			if got != tc.want {
+				t.Errorf("IsFatalError(%v) = %v; want %v", tc.err, got, tc.want)
+			}
+		})
+	}
+}
+
+func TestUnwrap(t *testing.T) {
+	err := cerrors.New("test error")
+	fatalErr := cerrors.NewFatalError(err)
```

Comment:
notices that all the tests are using `cerrors.New` as the input for `NewFatalError`, we should maybe have one case where it's a normal error, cuz not all error we're gonna have are `cerrors`

Reply:
@maha-hajja That was my thinking as well, and I tried it, but it seems we can't use that pkg at all. See https://github.com/ConduitIO/conduit/actions/runs/10596314041/job/29364014509#step:5:44.

Reply:
we could use `fmt.Errorf` instead, doesn't have to be the `errors` package, that should work right?

Reply:
@maha-hajja absolutely! I actually did that after I had left my comment 😅. You can find it [here](https://github.com/ConduitIO/conduit/blob/e812e1c64655e000cc6df350c3f05c7fb59d88aa/pkg/foundation/cerrors/fatal_test.go#L51).

Reply:
@maha-hajja I'm shipping this PR, because I'm assuming your only comment left was already addressed, but let me know if that wasn't the case, and I'll open a new PR.